client/data/courses/use-update-user-course-progression-mutation.js

```diff
@@ -18,9 +18,9 @@ function useUpdateUserCourseProgressionMutation( queryOptions = {} ) {
 			),
 		{
 			...queryOptions,
-			onSuccess( ...args ) {
-				queryClient.invalidateQueries( [ 'course-progression', args.courseSlug ] );
-				queryOptions.onSuccess?.( ...args );
+			onSuccess( data, variables, context ) {
+				queryClient.invalidateQueries( [ 'courses', variables.courseSlug ] );
```

Comment:
Invalidate the query to fetch the latest data so that we don't need to handle local completed status.